src/AutoRest.CSharp/Mgmt/Output/Resource.cs

```diff
@@ -127,8 +127,8 @@ protected internal Resource(IReadOnlyDictionary<OperationSet, IEnumerable<Operat
             yield return new FieldDeclaration(FieldModifiers, ResourceData.Type, DataFieldName);
         }
 
-        public Resource(IReadOnlyDictionary<OperationSet, IEnumerable<Operation>> allOperations, string resourceName, ResourceTypeSegment resourceType, ResourceData resourceData, BuildContext<MgmtOutputLibrary> context)
-            : this(allOperations, resourceName, resourceType, resourceData, context, ResourcePosition)
+        public Resource(IReadOnlyDictionary<OperationSet, IEnumerable<Operation>> allOperations, string resourceDataSchemaName, ResourceTypeSegment resourceType, ResourceData resourceData, BuildContext<MgmtOutputLibrary> context)
+            : this(allOperations, resourceDataSchemaName, resourceType, resourceData, context, ResourcePosition)
```

Comment:
Considering how this constructor is invoked, this `resourceDataSchemaName` now is really the name of this resource now. Could we change the name back?
```suggestion
            : this(allOperations, resourceName, resourceType, resourceData, context, ResourcePosition)
```